### PR TITLE
sorry for adding these into only one commit

### PR DESCRIPTION
**Pull request by lionelfeng**

### CODE REVIEW

File: fwang1/solutions/P202.py [hunk 1]
```diff
@@ -8,9 +8,7 @@ def isHappy(self, n):
         
         while n != 1:
             digit_char_list = []
-            digit_char_list.extend(str(n))
-            digit_char_list.sort()
-            
+            digit_char_list.extend(str(n))
```
Comment (updownlife) [inline fwang1/solutions/P202.py]: may need to consider the time cost of converting integer to string

File: fwang1/solutions/P020.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        dict = {
+            ')':'(',
+            ']':'[',
+            '}':'{'
+            }
+            
+        stack = []
+        
+        for i in range(len(s)):
+            if s[i] in dict.values():
```
Comment (updownlife) [inline fwang1/solutions/P020.py]: It's O(n) here, can be reduced to O(1) by dict.get(key)

Comment (lionelfeng) [inline fwang1/solutions/P020.py]: how about this: 
``` python
 if s[i] in set(dict.values()):
```

File: fwang1/solutions/P202.py (hunk 1 above)
Comment (lionelfeng) [inline fwang1/solutions/P202.py]: interesting... how much time would it spent on the converting? from str to int and int to str？

File: fwang1/solutions/P020.py (hunk 2 above)
Comment (updownlife) [inline fwang1/solutions/P020.py]: I think set(dict.values()) is O(n)

Comment (lionelfeng) [inline fwang1/solutions/P020.py]: ok. I thought there were other non-bracket chars mixed in the input...

Comment (lionelfeng) [inline fwang1/solutions/P020.py]: but you did remind me that set(list) is not a good way to improve the performance.

Comment (updownlife) [inline fwang1/solutions/P020.py]: 😄

File: fwang1/solutions/P202.py (hunk 1 above)
Comment (updownlife) [inline fwang1/solutions/P202.py]: O(n) n is the length is string

File: fwang1/solutions/P094.py [hunk 3]
```diff
@@ -11,21 +11,19 @@ def inorderTraversal(self, root):
...
-            else:
-            	result.append(stack.pop())
-            	if cur.right:
-            		stack.append(cur.right)
 
-        return map(lambda x: x.val, result)
+        # niubi! - rewrote it from Steven's code.
+        while cur or stack:
+        	while cur:
+				stack.append(cur)
```
Comment (updownlife) [inline fwang1/solutions/P094.py]: haha, please revise the indent here

**Pull request merged**